backend/app/api/api_v1/endpoints/scheduler.py

```diff
@@ -50,11 +50,9 @@ def list_schedules(
 def create_schedule(
         dataset_id: str,
         schedule: Schedule,
+        dataset_repository: DatasetRepository = Depends(get_dataset_repository),
 ):
-    dataset = _resource_exists(
-        settings.DATASET_INDEX,
-        dataset_id,
-    )
+    dataset = get_by_key_or_404(dataset_id, dataset_repository)
 
     schedule = scheduler.add_schedule(
         schedule=schedule,
```

Comment:
Need to update `datasource_id=dataset["_source"]["datasource_id"],` to `datasource_id=dataset.datasource_id,`

Reply:
Oh, right!